code/modules/clothing/glasses/_glasses.dm

```diff
@@ -397,6 +397,29 @@
 	inhand_icon_state = "gar"
 	glass_colour_type = /datum/client_colour/glass_colour/red
 
+/obj/item/syndicate_contacts
+	name = "suspicious contact case"
+	desc = "A sinister red case that contains two shiny black contancts lenses."
+	w_class = WEIGHT_CLASS_TINY
+	icon = 'icons/obj/device.dmi'
+	icon_state = "contacts"
+
+/obj/item/syndicate_contacts/attack_self(mob/user, modifiers)
+	. = ..()
+	if(!user.get_organ_slot(ORGAN_SLOT_EYES))
+		to_chat(user, span_warning("You have no eyes to apply the contacts to!"))
+	var/obj/item/organ/internal/eyes/eyes = user.get_organ_slot(ORGAN_SLOT_EYES)
+
+	to_chat(user, span_notice("You begin applying the contact lenses to your eyes..."))
+	if(do_after(user, 30))
+		if(eyes)
+			to_chat(user, span_notice("the contacts seamlessly merge with your iris."))
+			eyes.flash_protect = FLASH_PROTECTION_WELDER
+			to_chat(user, span_warning("The [src] disintegrates into nothing."))
+			qdel(src)
+		else
+			to_chat(user, span_warning("You have no eyes to apply the contacts to!"))
+
```

Comment:
early returns and seconds defines

we can hopefully use the `target` argument of the `do_after` to cancel our progress bar when the eyes are removed, as it checks for the target's loc staying the same. so long as eyes register the player's head as their loc it should be fine (?)

```suggestion
	if(!do_after(user, 3 SECONDS, src))
		return
	to_chat(user, span_notice("The contacts seamlessly merge with your iris."))
	eyes.flash_protect = FLASH_PROTECTION_WELDER
	to_chat(user, span_warning("\The [src] disintegrates into nothing."))
	qdel(src)

```